source/Octopus.Tentacle/Kubernetes/KubernetesScriptPodCreator.cs

```diff
@@ -284,14 +288,33 @@ protected async Task<V1Container> CreateScriptContainer(StartKubernetesScriptCom
 
                     //We intentionally exclude setting "TentacleJournal" since it doesn't make sense to keep a Deployment Journal for Kubernetes deployments
                 },
-                Resources = new V1ResourceRequirements
+                Resources = resourceRequirements
+            };
+        }
+
+        V1ResourceRequirements GetScriptPodResourceRequirements()
+        {
+            var json = KubernetesConfig.PodResourceJson;
+            if (!string.IsNullOrWhiteSpace(json))
+            {
+                try
                 {
-                    //set resource requests to be quite low for now as the scripts tend to run fairly quickly
-                    Requests = new Dictionary<string, ResourceQuantity>
-                    {
-                        ["cpu"] = new("25m"),
-                        ["memory"] = new("100Mi")
-                    }
+                    return KubernetesJson.Deserialize<V1ResourceRequirements>(json);
+                }
+                catch (Exception e)
+                {
+                    //if we can't parse the JSON, fall back to the defaults below and warn the user
+                    log.WarnFormat(e, $"Failed to deserialize env.{KubernetesConfig.PodResourceJsonVariableName} into valid pod resource requirements. Using default values. Value: {json}");
```

Comment:
```suggestion
                    log.WarnFormat(e, $"Failed to deserialize env.{KubernetesConfig.PodResourceJsonVariableName} into valid pod resource requirements.\nJSON: {json}\nUsing default values.");
```
Perhaps something like that suggestion to make it clear that the `json` is the thing that cannot be deserialised rather than the default values being applied?